server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/controllers/IoBasedAdmissionController.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ratelimitting.admissioncontrol.controllers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.concurrency.OpenSearchRejectedExecutionException;
+import org.opensearch.node.NodeResourceUsageStats;
+import org.opensearch.node.ResourceUsageCollectorService;
+import org.opensearch.ratelimitting.admissioncontrol.enums.AdmissionControlActionType;
+import org.opensearch.ratelimitting.admissioncontrol.enums.AdmissionControlMode;
+import org.opensearch.ratelimitting.admissioncontrol.settings.IoBasedAdmissionControllerSettings;
+
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ *  Class for IO Based Admission Controller in OpenSearch, which aims to provide IO utilisation admission control.
+ *  It provides methods to apply admission control if configured limit has been reached
+ */
+public class IoBasedAdmissionController extends AdmissionController {
+    public static final String IO_BASED_ADMISSION_CONTROLLER = "global_io_usage";
+    private static final Logger LOGGER = LogManager.getLogger(IoBasedAdmissionController.class);
+    public IoBasedAdmissionControllerSettings settings;
+
+    /**
+     * @param admissionControllerName       name of the admissionController
+     * @param resourceUsageCollectorService instance used to get resource usage stats of the node
+     * @param clusterService                instance of the clusterService
+     */
+    public IoBasedAdmissionController(
+        String admissionControllerName,
+        ResourceUsageCollectorService resourceUsageCollectorService,
+        ClusterService clusterService,
+        Settings settings
+    ) {
+        super(admissionControllerName, resourceUsageCollectorService, clusterService);
+        this.settings = new IoBasedAdmissionControllerSettings(clusterService.getClusterSettings(), settings);
+    }
+
+    @Override
+    public boolean isEnabledForTransportLayer(AdmissionControlMode admissionControlMode) {
```

Comment:
Just dont register this AC in case not linux. We dont need to add this method.

Reply:
Agree